### ISSUE TITLE
Grand Plan

### ISSUE THREAD

**Issue by Deraen** [cljsjs/boot-cljsjs#3]
Thought I would write down what I proposed on IRC,

Because not every JS lib will ever be packaged for Cljsjs there should be an easy way to use stuff from Webjars and any other jars which might contains js files.

So in order of preference user could use:
1. Cljsjs packaged library
2. Webjars
3. Random jar

The only thing that needs to happen for boot-cljs to use any library is that fileset contains .inc.js, .ext.js or .lib.js when Cljs task is run. This should be easy to achieve with one boot plugin.
1. `(from-cljsjs :target "public")`
   - `[cljsjs/react "0.12.2"]`
   - This task would go through classpath (all jars) and search for any *.[inc|ext|lib].js files and copy then to the target path.
   - If/when boot tasks have way to find out whats the docroot, that could be used
2. `(from-webjars :name "momentjs/moment.js" :target "public/moment.inc.js")`
   - `[org.webjars/momentjs "2.8.3"]`
   - Like 3. but files in webjars have long paths and user wouldn't want to write e.g. "META-INF/resources/webjars/momentjs/2.8.3/moment.js"
   - Webjars locator can be used to list files from webjars in classpath
     - Might be even easier to use the same code as 1. and just match found files to  webjars path regex.
   - And then some clojure code can be used to find proper resource for short name
   - https://github.com/Deraen/boot-from-jars/blob/master/src/deraen/boot_from_jars/impl.clj
3. `(from-jars :path "react/react.js" :target "public/react.inc.js")`
   - `[com.facebook/react "0.12.2"]`
   - User might want to use file from e.g. vendor provided jar
   - User provides full path to file in classpath and the file is copied into target
- Feel free to propose better names for these tasks
- In addition to being directly usable, 2. and 3. might be useful for packaging stuff to Cljsjs
- Target path is meaningful only in `:optiomization :none` mode because there the webserver needs to serve these files.

Implementation for 2. and 3. can be found on my repo: https://github.com/Deraen/boot-from-jars
I can do PR to merge them here.

Implementation for 1. is mostly the same as existing `import-js` here but without concatenation (which should be separate task).

Some notes about implementation:
- `import-js` I think currently runs every time watch is triggered
- my tasks run only once
- Especially going through all jars is going to be slow (1. and 2.) and should only be ran rarely
- It is possible to add dependencies at run time using set-env etc.
  - We could check if `java.class.path` has changed after previous run?
- In production build the target path has no meaning and the files need not be present on .jar. Thus `add-source` should be used instead of `add-resource` to add the files to fileset.
  - `:production true` option for each task? is there any better way?
- Instead of having task which run multiple times to add several files, would it make sense the have task that takes multiple paths as argument?
